client-bagel-buddy/src/components/header/Header.js

```diff
@@ -22,6 +23,7 @@ function Header() {
                 if (res.success) {
                     let { data } = res;
                     setUser(data);
+                    setNewBio(user.bio);
```

Comment:
setNewBio(data.bio)

because setUser needs some time to update the user data. The moment we called `user.bio` here the user may still be null (not fully updated yet)